src/shared/watchdog.c

```diff
@@ -115,8 +118,18 @@ int watchdog_set_timeout(usec_t *usec) {
 }
 
 int watchdog_ping(void) {
+        usec_t ntime;
         int r;
 
+        ntime = now(clock_boottime_or_monotonic());
+
+        if (_unlikely_(watchdog_last_ping == USEC_INFINITY)) {
+                watchdog_last_ping = ntime;
+        } else {
+                if ((ntime - watchdog_last_ping) < (watchdog_timeout / 2))
```

Comment:
consider adding an `assert(ntime >= watchdog_last_ping);` in the line before, to document that an underflow is not possible unless time goes backwards